sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -147,33 +147,15 @@ using KernelFunc = std::function<void(const sycl::nd_item<NDims> &)>;
 
 // Struct to wrap dimension info and lambda function to be invoked by
 // CM Kernel launcher that only accepts raw function pointer for
-// kernel execution. Function instances of 'InvokeLambda' un-wrap this
-// struct instance and invoke lambda function ('Func')
-template <int NDims> struct LambdaWrapper {
+// kernel execution. Function instances of 'KernelWrapper' un-wrap
+// this struct instance and invoke lambda function ('Func')
+template <int NDims> struct KernelWrapperData {
```

Comment:
I don't see how KernelWrapper is better than InvokeLambda. If Lambda does not cover all kernel cases then InvokeKernel or KernelInvoker?
Then KernelWrapperData -> KernelInvocationContext.

If this all is only supposed to be used on a host device (and/or esimd emulator), then I'd add 'Host' prefix as well.

Reply:
Warning: naming discussion.
Ok, the point was to make more sense (and less code). Yes, kernel is not always lambda.

There are many ways to do it. I consider KernelWrapper and KernelWrapperData clear enough and find adding "Host" prefix for things used inside esimd_emulator only an overkill and not bringing more sense.

If inevitable, the naming can be changed **to the required liking**.
e.g.,

InvokeKernel
KernelInvokationContext

would be totally clear.

Reply:
Classes should not be named with verbs, but adhjectives. KernelCaller, KernelInvoker

> Warning: naming discussion.
Not sure I understand the warning. You started changing names, so I commented.

Reply:
Konstantin, InvokeKernel is a function (template), not a class.

P.S. regarding classes, how they should be named (verbs/adjectives/nouns) is a debatable taking into account that they can be callable in C++ (as we all know). In other languages, e.g. JavaScript, everything is an object. Should we name everything in JavaScript not using verbs? I doubt that.